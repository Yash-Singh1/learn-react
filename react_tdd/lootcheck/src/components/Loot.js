import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/actions';

export class Loot extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
  }

  computeBitcoin() {
    const { bitcoin } = this.props;

    if (bitcoin.bpi) {
      return this.props.balance / bitcoin.bpi.USD.rate_float;
    } else {
      return '';
    }
  }

  render() {
    return (
      <div>
        <h3>Bitcoin balance: {this.computeBitcoin()}</h3>
      </div>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return state;
}

export default connect(mapStateToProps, { fetchBitcoin })(Loot);
