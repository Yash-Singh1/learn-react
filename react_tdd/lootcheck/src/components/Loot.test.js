import React from 'react';
import { mount, shallow } from 'enzyme';
import { Loot } from './Loot';

describe('Loot', () => {
  let props = {
    balance: 10,
    bitcoin: {}
  };
  let loot = shallow(<Loot {...props} />);

  it('renders properly', () => {
    expect(loot.instance()).toMatchSnapshot();
  });

  describe('when mounted', () => {
    const mockFetchBitcoin = jest.fn();

    beforeEach(() => {
      props.fetchBitcoin = mockFetchBitcoin;
      loot = mount(<Loot {...props} />);
    });

    it('dispatches the `fetchBitcoin()` method it recieves from props', () => {
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });

    describe('when there are valid bitcoin props', () => {
      beforeEach(() => {
        props = {
          balance: 10,
          bitcoin: { bpi: { USD: { rate_float: 1000 } } }
        };
        loot = shallow(<Loot {...props} />);
      });

      it('displays the correct bitcoin value', () => {
        expect(loot.find('h3').text()).toEqual('Bitcoin balance: 0.01');
      });
    });
  });
});
