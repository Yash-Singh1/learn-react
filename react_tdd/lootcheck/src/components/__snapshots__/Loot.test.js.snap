// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `
Loot {
  "context": Object {},
  "props": Object {
    "balance": 10,
    "bitcoin": Object {},
  },
  "refs": Object {},
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <Loot
        balance={10}
        bitcoin={Object {}}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div>
        <h3>
          Bitcoin balance: 
          
        </h3>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
